tests/phpunit/tests/blocks/register.php

```diff
@@ -192,6 +205,52 @@ public function test_generate_block_asset_handle_core_block() {
 		);
 	}
 
+	/**
+	 * @ticket 60233
+	 */
+	public function test_generate_block_asset_handle_core_block_module() {
+		$block_name = 'core/paragraph';
+
+		$this->assertSame(
+			'wp-block-paragraph-editor-script-module',
```

Comment:
Should it use the same pattern as core blocks do? Examples:

https://github.com/WordPress/wordpress-develop/blob/27087b71e0c109af68dbc0545893217fc80350db/src/wp-includes/blocks/file.php#L67

https://github.com/WordPress/wordpress-develop/blob/27087b71e0c109af68dbc0545893217fc80350db/src/wp-includes/blocks/image.php#L327

How important is the module ID here? The name `generate_block_asset_handle` is also a bit misleading here, as the helper generates a module id rather than a handle.

Reply:
The script module ID is only important as a unique identifier analogous to script handles. I don't think that it's important for generated core script module IDs to follow that pattern.

That ID was introduced in https://github.com/WordPress/gutenberg/pull/56143, I don't see discussion there on the naming choice. I suspect it was a convenient and clear identifier that should be unique.

We could try to follow a similar pattern for core blocks, but I'm not sure it's worth it. 

There is one reason to maintain the same form. If we decide to switch to use `viewScriptModule` for core blocks, I'm not sure whether it would be a breaking change to change the script module ID.

---

> the name `generate_block_asset_handle` is also a bit misleading here

I agree, but the function serves just as well for module IDs (if we keep it as-is and don't change the form). We can't change an existing function name, so it seemed preferable to just reuse it and not introduce another function to do the same thing.

Reply:
Yes, the script module identifier is not important for enqueued script modules that are not meant to be dependencies of other script modules (i.e., script modules without exports) because as they don't export anything, there's no need to match the module identifier with the npm package to get the types in the IDE.

It can still be useful, although I don't expect it to be common. For example, in case one script module wants to make sure that another script module has loaded because it needs it for some reason (like reading/writing to the `store()` initialized in that script module), it can ensure that it is loaded by importing that script module identifier.

I don't have a strong opinion about the naming though. I used `@wordpress/block-library/block-name` to follow a familiar npm pattern, but whatever you decide is fine.

Reply:
`wp-block-block-name-view-script-module` is also fine if that's more aligned with the rest of WordPress handles. People can do:

```js
store("myPlugin", {
  actions: {
    *someAction() {
      // Make sure that the core/navigation store has loaded.
      yield import("wp-block-navigation-view-script-module");

      const { actions } = store("core/navigation");
      actions.openMenu();
    },
  },
});
```